Source/Scene/ClassificationModel.js

```diff
@@ -0,0 +1,1869 @@
+            var attributeLocations = shader.attributeLocations;
+
+            var pickUniformMap;
+            var pickShader = rendererPickPrograms[technique.program];
+            var pickVertexShaderSource = pickShader.vertexShaderSource;
+            var pickFragmentShaderSource = pickShader.fragmentShaderSource;
+
+            if (defined(model._pickUniformMapLoaded)) {
+                pickUniformMap = model._pickUniformMapLoaded(uniformMap);
+            } else {
+                // This is unlikely, but could happen if the override shader does not
+                // need new uniforms since, for example, its pick ids are coming from
+                // a vertex attribute or are baked into the shader source.
+                pickUniformMap = combine(uniformMap);
+            }
+
+            var nodeCommand = new Vector3DTilePrimitive({
+                classificationType : model._classificationType,
+                positions : positionsBuffer,
+                indices : indices,
+                indexOffsets : indexOffsets,
+                indexCounts : indexCounts,
+                batchIds : batchIds,
+                vertexBatchIds : vertexBatchIds,
+                batchedIndices : batchedIndices,
+                batchTable : batchTable,
+                boundingVolume : new BoundingSphere(), // updated in update()
+                _vertexShaderSource : vertexShaderSource,
+                _fragmentShaderSource : fragmentShaderSource,
+                _attributeLocations : attributeLocations,
+                _pickVertexShaderSource : pickVertexShaderSource,
+                _pickFragmentShaderSource : pickFragmentShaderSource,
+                _uniformMap : uniformMap,
+                _pickUniformMap : pickUniformMap,
+                _modelMatrix : new Matrix4(), // updated in update()
+                _boundingSphere : boundingSphere // used to update boundingVolume
+            });
+            runtimeNode.commands.push(nodeCommand);
+            nodeCommands.push(nodeCommand);
+        }
+    }
+
+    function createRuntimeNodes(model, context) {
+        var loadResources = model._loadResources;
+
+        if (!loadResources.finishedEverythingButTextureCreation()) {
+            return;
+        }
+
+        if (!loadResources.createRuntimeNodes) {
+            return;
+        }
+        loadResources.createRuntimeNodes = false;
+
+        var gltf = model.gltf;
+        var nodes = gltf.nodes;
+        if (nodes.length > 1) {
+            throw new RuntimeError('Only one node is supported when using a batched 3D tileset for classification.');
+        }
+
+        var gltfNode = nodes[0];
+        var runtimeNode = model._runtimeNode;
+        if (defined(gltfNode.matrix)) {
+            runtimeNode.matrix = Matrix4.fromColumnMajorArray(gltfNode.matrix);
+        } else {
+            // TRS converted to Cesium types
+            var rotation = gltfNode.rotation;
+            runtimeNode.translation = Cartesian3.from
```

Comment:
Seems like once this is trimmed down for one node/primitive, there are some obvious things that can be factored into a common file such as the quantization functions.  I wonder if extension handling can also be made more simple here since only quantization and RTC need to be supported.

Also remind me, can we completely ignore the shaders since we are assuming no alpha texture / discard?

Reply:
RTC, binary glTF, and quantization are the extensions that are supported. The other glTF extensions are for the material and technique which we don't even parse.

We now also completely ignore the shaders.